.github/workflows/release.yml

```diff
@@ -0,0 +1,182 @@
+name: Ion Python Release
+
+on:
+  push:
+    branches: [ master ]
+    tags:
+      - "v*.*.*"
+
+jobs:
+  test:
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.7', '3.8', '3.9', '3.10', 'pypy-3.6', 'pypy-3.7', 'pypy-3.8' ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up python
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Create a virtual environment
+        run: git submodule init && git submodule update && python3 -m venv ./venv && . venv/bin/activate
+      - run: pip install -r requirements.txt
+      - run: pip install -e .
+      - run: py.test
+
+  source-distribution:
+    if: startsWith(github.ref, 'refs/tags/')
+    name: Source distribution
+    runs-on: ubuntu-latest
+    needs: [test]
+    strategy:
+      matrix:
+        python-version: [ '3.10' ]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build sdist
+        run: python3 setup.py sdist
+
+      - name: Upload source to Github
```

Comment:
Where is this uploading to? A release?

Reply:
We upload source to GitHub Actions so we can install the built source/wheels directly from Github. E.g, we upload test-driver output to [ion-java GHA](https://github.com/amzn/ion-java/actions/runs/2099291871) so users can download the output when workflow fails, you will find them in the artifacts section at the bottom of the page.

The reason we upload them to Github is because later on we will [download](https://github.com/amzn/ion-python/pull/199/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R141-R143) them from GitHub and zip all wheels in a separate job and upload them to AWS. Since we built wheels on different instances with different OS so I tried to avoid different upload behaviors (E.g. `zip` command works on Mac but Windows use some other command). Therefore I try to gather all wheels in one instance and release them in case some wheels are lost during the release process.